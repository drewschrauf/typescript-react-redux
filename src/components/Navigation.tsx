import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const Wrapper = styled.div`
  margin-bottom: 10px;
`;

const LinkList = styled.ul`
  position: absolute;
  border: 1px solid black;
  background-color: papayawhip;

  padding: 10px;
  width: 70px;
  margin: 0;

  visibility: hidden;
  opacity: 0;
  transition: all 0.3s ease;
`;

const LinkListItem = styled.li`
  list-style: none;
`;

const LinkWrapper = styled.div`
  position: relative;
  display: inline-block;
  margin-right: 5px;

  &:hover ${LinkList} {
    visibility: visible;
    opacity: 1;
  }
`;

const Navigation = () => (
  <Wrapper>
    <LinkWrapper>
      <Link to="/">Countâ–¼</Link>
      <LinkList>
        <LinkListItem>
          <Link to="/by/1">By 1</Link>
        </LinkListItem>
        <LinkListItem>
          <Link to="/by/2">By 2</Link>
        </LinkListItem>
        <LinkListItem>
          <Link to="/by/3">By 3</Link>
        </LinkListItem>
        <LinkListItem>
          <Link to="/by/5">By 5</Link>
        </LinkListItem>
        <LinkListItem>
          <Link to="/by/8">By 8</Link>
        </LinkListItem>
      </LinkList>
    </LinkWrapper>
    <LinkWrapper>
      <Link to="/about/">About</Link>
    </LinkWrapper>
  </Wrapper>
);
export default Navigation;
