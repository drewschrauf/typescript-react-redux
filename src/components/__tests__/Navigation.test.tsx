import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../Navigation';

it('should render navigation', () => {
  const root = render(
    <Router>
      <Navigation />
    </Router>,
  );
  expect(root.container.innerHTML).toMatchSnapshot();
});
