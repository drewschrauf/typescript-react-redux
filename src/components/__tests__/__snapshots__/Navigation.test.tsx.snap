// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render navigation 1`] = `"<div class=\\"Navigation__Wrapper-j6h1ss-0 esetmv\\"><div class=\\"Navigation__LinkWrapper-j6h1ss-3 gbXNJi\\"><a href=\\"/\\">Countâ–¼</a><ul class=\\"Navigation__LinkList-j6h1ss-1 jVRZQz\\"><li class=\\"Navigation__LinkListItem-j6h1ss-2 kkyviC\\"><a href=\\"/by/1\\">By 1</a></li><li class=\\"Navigation__LinkListItem-j6h1ss-2 kkyviC\\"><a href=\\"/by/2\\">By 2</a></li><li class=\\"Navigation__LinkListItem-j6h1ss-2 kkyviC\\"><a href=\\"/by/3\\">By 3</a></li><li class=\\"Navigation__LinkListItem-j6h1ss-2 kkyviC\\"><a href=\\"/by/5\\">By 5</a></li><li class=\\"Navigation__LinkListItem-j6h1ss-2 kkyviC\\"><a href=\\"/by/8\\">By 8</a></li></ul></div><div class=\\"Navigation__LinkWrapper-j6h1ss-3 gbXNJi\\"><a href=\\"/about/\\">About</a></div></div>"`;
